LibreNMS/OS/Iosxe.php

```diff
@@ -43,10 +54,95 @@ class Iosxe extends Ciscowlc implements
     WirelessRsrpDiscovery,
     WirelessSnrDiscovery
 {
+    use SyncsModels;
     use CiscoCellular;
 
     public function pollOS(): void
     {
         // Don't poll Ciscowlc FIXME remove when wireless-controller module exists
     }
+
+    protected $isis_codes = [
+        'l1IntermediateSystem' => 'L1',
+        'l2IntermediateSystem' => 'L2',
+        'l1L2IntermediateSystem' => 'L1L2',
+        'unknown' => 'unknown',
+    ];
+
+    public function discoverIsIs(): Collection
+    {
+        // Check if the device has any ISIS enabled interfaces
+        $circuits = snmpwalk_cache_oid($this->getDeviceArray(), 'CISCO-IETF-ISIS-MIB::ciiCirc', []);
+        $adjacencies = new Collection;
+
+        if (! empty($circuits)) {
+            $adjacencies_data = snmpwalk_cache_twopart_oid($this->getDeviceArray(), 'CISCO-IETF-ISIS-MIB::ciiISAdj', [], null, null, '-OQUstx');
+            $ifIndex_port_id_map = $this->getDevice()->ports()->pluck('port_id', 'ifIndex');
+
+            // No ISIS enabled interfaces -> delete the component
+            foreach ($circuits as $circuit_id => $circuit_data) {
+                if (! isset($circuit_data['ciiCircIfIndex'])) {
+                    continue;
+                }
+
+                if ($circuit_data['ciiCircPassiveCircuit'] == 'true') {
+                    continue; // Do not poll passive interfaces
+                }
+
+                $adjacency_data = Arr::last($adjacencies_data[$circuit_id] ?? [[]]);
+
+                $attributes = [
```

Comment:
Instead of dealing with arrays, make a model object first thing:
`$adjacency = new IsisAdjacency([ ... ]);`